nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,249 @@ def save(self, *args, **kwargs):
     def delete(self, *args, **kwargs):
         # TODO: should we behave like CustomFieldChoice and raise a ProtectedError if this choice is in use anywhere?
         super().delete(*args, **kwargs)
+
+
+class ObjectMetadataManager(BaseManager.from_queryset(RestrictedQuerySet)):
+    use_in_migrations = True
+
+    def get_for_model(self, instance):
+        """Return all ObjectMetadatas assigned to the given instance."""
+        cache_key = f"{self.get_for_model.cache_key_prefix}.{instance}"
+        queryset = cache.get(cache_key)
+        if queryset is None:
+            assigned_object_type = ContentType.objects.get_for_model(instance)
+            assigned_object_id = instance.pk
+            queryset = self.get_queryset().filter(
+                assigned_object_type=assigned_object_type, assigned_object_id=assigned_object_id
+            )
+            cache.set(cache_key, queryset)
```

Comment:
If we're going to do caching here (not sure there's as much value given that this is per-instance rather than per-model-class), then you need to make sure the cache gets cleared appropriately when creating, updating, or deleting any ObjectMetadata records. Might not be worth the trouble here?

Reply:
Agreed removed the caching feature from this helper function.